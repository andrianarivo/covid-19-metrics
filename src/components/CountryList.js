import PropTypes from 'prop-types';
import CountryItem from './CountryItem';

export default function CountryList(props) {
  const { countries } = props;
  return (
    <table>
      {countries.map((item) => (
        <CountryItem
          key={item.country}
          country={item.country}
          active={item.active}
          updated={item.updated}
        />
      ))}
    </table>
  );
}

CountryList.propTypes = {
  countries: PropTypes.arrayOf(Object).isRequired,
};
