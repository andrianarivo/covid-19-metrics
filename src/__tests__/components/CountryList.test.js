import { create } from 'react-test-renderer';
import CountryList from '../../components/CountryList';

describe('Testing CountryList component', () => {
  test('renders correctly', () => {
    const countryItems = [
      {
        updated: 1687957193344,
        country: 'Afghanistan',
        active: 12338,
      },
      {
        updated: 1687957193333,
        country: 'Albania',
        active: 1696,
      },
    ];
    const tree = create(<CountryList countries={countryItems} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
