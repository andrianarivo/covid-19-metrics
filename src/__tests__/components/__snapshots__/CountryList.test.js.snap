// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CountryList component renders correctly 1`] = `
<table
  className="sc-kIDzkD gIEPdf"
>
  <tbody>
    <tr
      className="sc-jzuGiw YszEF"
    >
      <td>
        Afghanistan
      </td>
      <td>
        <ul>
          <li>
            12338
          </li>
          <li>
            a day ago
          </li>
        </ul>
      </td>
    </tr>
    <tr
      className="sc-jzuGiw YszEF"
    >
      <td>
        Albania
      </td>
      <td>
        <ul>
          <li>
            1696
          </li>
          <li>
            a day ago
          </li>
        </ul>
      </td>
    </tr>
  </tbody>
</table>
`;
